extends Node

var api_key = ""
var app_id = "49ab8a790c11476588d9dd4746b745bb"
var url = "https://dashscope.aliyuncs.com/api/v1/apps/%s/completion" % app_id
@onready var http_request = $HTTPRequest  # è·å– HTTPRequest èŠ‚ç‚¹
@onready var responsebox = $"../ResponseBox"
func _ready():
	if http_request == null:
		print("http_request is null")
		return
	http_request.request_completed.connect(_on_request_completed)


func set_api_key(key: String):
	api_key = key
	#print("key here", api_key)

func send_message(message: String):
	#print("å‘é€æ¶ˆæ¯åˆ° API: ", message)
	responsebox.show()
	
	var headers = [
		"Authorization: Bearer " + api_key,
		"Content-Type: application/json"
	]
	
	var body = {
		"input": {
			"prompt": message  # è¿™é‡Œä¼ å…¥åŠ¨æ€æ¶ˆæ¯
		},
		"parameters": {},
		"debug": {}
	}

	var json_body = JSON.stringify(body)

	var error = http_request.request(url, headers, HTTPClient.METHOD_POST, json_body)
	if error != OK:
		print("HTTP è¯·æ±‚å¤±è´¥ï¼Œé”™è¯¯ä»£ç : ", error)

# å¤„ç† API å“åº”
func _on_request_completed(result, response_code, headers, body):
	var data = body.get_string_from_utf8()
	#print("ğŸ” åŸå§‹è¿”å›æ•°æ®ï¼š", data)
	var response = JSON.parse_string(data) # Returns null if parsing failed.
	response = response["output"]["text"]
	print(response)
	responsebox.text = response
